src/TestFramework/Extension.WinUI/UITestMethodAttribute.cs

```diff
@@ -81,5 +110,108 @@ public override TestResult[] Execute(ITestMethod testMethod)
 
             return new TestResult[] { result };
         }
+
+        private static Type GetApplicationType(Assembly assembly)
+        {
+            var attribute = assembly.GetCustomAttribute<WinUITestTargetAttribute>();
+            if (attribute == null || attribute.ApplicationType == null)
+            {
+                return null;
+            }
+
+            return attribute.ApplicationType;
+        }
+
+        private static DispatcherQueue GetApplicationDispatcherQueue(Assembly assembly)
+        {
+            if (applicationDispatcherQueue != null)
+            {
+                return applicationDispatcherQueue;
+            }
+
+            if (isApplicationInitialized)
+            {
+                return null;
+            }
+
+            var applicationType = GetApplicationType(assembly);
+            if (applicationType == null)
+            {
+                return null;
+            }
+
+            // We need to initialize the SDK before calling Application.Start
+            try
+            {
+                // We need to execute all module initializers before doing any WinRT calls.
+                // This will cause the [ModuleInitialzer]s to execute, if they haven't yet.
+                var id = applicationType.Assembly.GetType("Microsoft.WindowsAppSDK.Runtime.Identity");
+                if (id != null)
+                {
+                    _ = Activator.CreateInstance(id);
+                }
+            }
+            catch
+            {
+            }
+
+            return InitializeApplication(applicationType);
+        }
+
+        private static DispatcherQueue GetDispatcherQueue(Assembly assembly)
+        {
+            if (DispatcherQueue != null)
+            {
+                return DispatcherQueue;
+            }
+
+            if (GetApplicationDispatcherQueue(assembly) is { } appDispatcherQueue)
+            {
+                return appDispatcherQueue;
+            }
+
+            try
+            {
+                if (Window.Current?.DispatcherQueue is { } windowDispatcherQueue)
+                {
+                    return windowDispatcherQueue;
+                }
+            }
+            catch
+            {
+            }
+
+            return null;
+        }
+
+        private static DispatcherQueue InitializeApplication(Type applicationType)
```

Comment:
Just wondering if this should be a `void` method simply doing the app initialization and the caller would then return `applicationDispatcherQueue`.